### PR TITLE
Jenkins2

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
This PR test Jenkins upgrade https://trello.com/c/bfzIv5Pj/2428-ci-master-jenkins-upgrade :

- upgrade Jenkins to 2.32.1 (preparation for 2.32.2) 
- upgrade all the plugins to latest version
- secure jenkins with admin account by default
- blue is green !

**NOTE: that I wasn't successful with credentials for nodes https://trello.com/c/eKkIBg0r/1854-add-an-admin-password-for-devspace-jenkins#comment-588cdf1327142b86bbc8a2c0.**

cc @sbesson @joshmoore 

deployed in https://10.0.51.154:8443/ using https://github.com/openmicroscopy/infrastructure/pull/245

### CODE REVIEW

**Comment by aleksandra-tarkowska** [comment]
@sbesson new version doesnt like BF-ant:

```
You have data stored in an older format and/or unreadable data.

It is acceptable to leave unreadable data in these files, as Jenkins will safely ignore it. To avoid the log messages at Jenkins startup you can permanently delete the unreadable data by resaving these files using the button below.

Type	Name	Error
hudson.model.FreeStyleProject	BIOFORMATS-ant	CannotResolveClassException: hudson.tasks.Ant
hudson.model.Hudson		CannotResolveClassException: hudson.markup.RawHtmlMarkupFormatter
```

**Comment by aleksandra-tarkowska** [comment]
tested with https://github.com/openmicroscopy/infrastructure/pull/245

**Comment by aleksandra-tarkowska** [comment]
Looks like all works, the only issue at the moment is manual groovy script approval. I am not sure why `home/scriptApproval.xml` is not enough.

**Comment by joshmoore** [comment]
Note: on that instance, I've tried adding "stage" sections to the `Trigger` job per the UI request:

```
stage('bioformats') {
  ...
}
```

**Comment by joshmoore** [comment]
@aleksandra-tarkowska : do you think this upgrade process is going to be applicable to ci-master? Can you tell if there's any risk of logs, etc. being lost?

**Comment by aleksandra-tarkowska** [comment]
I never had opportunity to deal with ci-master deployment.

**Comment by joshmoore** [comment]
Testing on Ola's version seems fine, and changes make sense to me, though I still need to test an upgrade on a copy of idr-ci.

@sbesson : @aleksandra-tarkowska proposes to consider `master` as a new breaking version and we branch off from `0.4.0` if any other minor fixes are needed. That would allow us to add a small number of other fixes here (e.g. swarm --> jnlp) before cutting the next major version (0.5.0, 0.9.0, or even 1.0.0)

**Comment by joshmoore** [comment]
Still looking into potential issues; currently needed devspaces should be using 0.3.x.

File: home/jobs/BIOFORMATS-ant/config.xml [hunk 1]
```diff
@@ -22,12 +22,12 @@
         <hudson.model.StringParameterDefinition>
           <name>testng.runWriterSaveBytesTests</name>
           <description></description>
-          <defaultValue>50</defaultValue>
+          <defaultValue>0</defaultValue>
```
Comment (aleksandra-tarkowska) [inline home/jobs/BIOFORMATS-ant/config.xml]: @sbesson looks like now it is not crashing

**Comment by aleksandra-tarkowska** [comment]
So, I think this is ok to go in. There were no more crashes since reducing tests to 0

**Comment by joshmoore** [comment]
Merging as the next "major" development branch (0.5.0-m1)

**Pull request merged**